### DISCUSSION TITLE
Why moving to another page doesn't run server/middleware ?

### DISCUSSION THREAD

**Discussion by kevhof** [nuxt/nuxt#18162] [Questions] [answered]
Let's imagine you are on the index page. You click a router-link to move to another page. The new page javascript is downloaded and then the page is displayed. Why aren't the serverMiddlewares executed during this step ? 

I know that the app is already in the browser cache but the new page has to be downloaded from somewhere right ?

**Comment by technicallypete** [comment, marked as answer]
I just started looking at using nuxt yesterday..

But, I would say that server/middleware is meant for the server/api only, not to be confused with SSR/universal rendering. Also, if using router links across the app, i don't believe the browser is actually making calls to the server to render the page -- it's just loading components from a JS bundle and fetching api data at that point.

SSR only runs when you visit the first page or manually reload the page.

**Reply by kevhof** [reply]
Thanks!
